datastore/datastore.go

```diff
@@ -0,0 +1,322 @@
+			"datastoreQuery.projectId":           projectID,
+			"datastoreQuery.timestamp":           k.Timestamp.Unix(),
+		})
+	}
+	return datastoreData
+}
+
+func fatalIfErr(log *logrus.Logger, err error) {
+	if err != nil {
+		log.WithError(err).Fatal("can't continue")
+	}
+}
+
+// ConnectDatastore establishes a datastore.Client from a base64 encoding JSON credentials file.
+func ConnectDatastore(base64Config string) (*datastore.Client, string, error) {
+	jsonConfig, err := base64.StdEncoding.DecodeString(base64Config)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to decode datastore credentials: %v", err)
+	}
+	projectId, err := jsonparser.GetString(jsonConfig, "project_id")
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to determine project_id from credentials file: %v", err)
+	}
+
+	ctx := context.Background()
+	creds, err := google.CredentialsFromJSON(ctx, jsonConfig, datastore.ScopeDatastore)
+	if err != nil {
+		return nil, "", err
+	}
+	c, err := datastore.NewClient(ctx, projectId, option.WithCredentials(creds))
+	if err != nil {
+		return nil, "", err
+	}
+	return c, projectId, err
+}
+
+//StackdriverResp gets the data of the wanted metric from the stackdriver API. Start time is set at -3 minutes to act as a Timestamp
+//as data from stackdriver is always 3 minutes old and refreshed every 1 minute. This timing also ensures we only ever get 1 point
+//back at a time.
+func StackdriverResp(projectId string, metric string) (*monitoring.ListTimeSeriesResponse, error) {
+	ctx := context.Background()
+	jsonConfig, err := base64.StdEncoding.DecodeString(base64Config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode datastore credentials: %v", err)
+	}
+
+	projectId, err = jsonparser.GetString(jsonConfig, "project_id")
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine project_id from credentials file: %v", err)
+	}
+
+	creds, err := google.CredentialsFromJSON(ctx, jsonConfig, monitoring.MonitoringScope)
+
+	hc := oauth2.NewClient(ctx, creds.TokenSource)
+
+	s, err := monitoring.New(hc)
+	if err != nil {
+		return nil, err
+	}
+
+	startTime := time.Now().UTC().Add(time.Minute * -3)
+	endTime := time.Now().UTC()
+
+	resp, err := s.Projects.TimeSeries.List(projectResource(projectId)).
+		Filter(fmt.Sprintf("metric.type=\"%s\"", metric)).
+		IntervalStartTime(startTime.Format(time.RFC3339)).
+		IntervalEndTime(endTime.Format(time.RFC3339)).
+		Do()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+
+}
+
+//StackdriverData converts a ListTimeSeriesResponse to a []map[string]interface{} to be used for the final output to be sent to new relic
+func StackdriverData(resp *monitoring.ListTimeSeriesResponse) ([]map[string]interface{}, error) {
+	var stackdriverMetricBody StackdriverMetric
+	var metricResult []map[string]interface{}
+
+	b, err := json.Marshal(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(b, &stackdriverMetricBody)
+
```

Comment:
extra space here. Also if you like these line can be combined, `if err := json.Unmarshal(b, &stackdriverMetricBody); err != nil {`